app/src/main/java/com/github/geohunt/app/model/DataPool.kt

```diff
@@ -0,0 +1,20 @@
+package com.github.geohunt.app.model
+
+class DataPool<K, T>(private val factory: (K) -> T) {

```

Comment:
I know about this, if you look up in the commit history you will see that this class was remove for this reason then reintroduced. It was reintroduced because of a bug, the `withDefault` method do not actually override the given key in the hash map. As such the bug I fixed was that went requesting multiple time an non existent key, the `withDefault` will call the factory method each time and won't actually insert the value in the HashMap. We could definitely do this for each instance of the HashMap but this is error prone. Having a single class to handle this a cleaner approach to this issue. Beside the use of a HashMap is an implementation details that should be hidden to the user